lib/cache-manager.js

```diff
@@ -0,0 +1,210 @@
+/* eslint-disable import/no-dynamic-require */
+/* eslint-disable global-require */
+
+'use strict';
+
+const logger = require('@janiscommerce/logger');
+const path = require('path');
+const CacheManagerError = require('./cache-manager-error');
+
+const STR = {
+	memory: {
+		dependency: 'lru-cache',
+		private: '_memory',
+		route: './memory-manager'
+	},
+	redis: {
+		dependency: 'redis',
+		private: '_redis',
+		route: './redis-manager'
+	}
+};
+
+/**
+ * CacheMAnager Class
+ */
+
+class CacheManager {
+
+	set clientPrefix(clientPrefix) {
+		this._clientPrefix = clientPrefix;
+	}
+
+	get clientPrefix() {
+		return this._clientPrefix;
+	}
+
+	get memory() {
+		this.init('_memory');
+		return this._memory;
+	}
+
+	set memory(memo) {
+		this._memory = memo;
+	}
+
+	get redis() {
+		this.init('_redis');
+		return this._redis;
+	}
+
+	set redis(red) {
+		this._redis = red;
+	}
+
+	/**
+   * Initialize CacheManager.
+   * @param {string} clientPrefix Name of the clientPrefix-prefix
+   */
+	constructor(clientPrefix) {
+		this.clientPrefix = this.validClientPrefix(clientPrefix);
+		this.memory = null;
+		this.redis = null;
+		logger.info(`Cache - cliente prefix: ${this.clientPrefix}`);
+	}
+
+	/**
+	 * Check the client-prefix
+	 * @param {String} prefix name of client-prefix.
+	 * @returns {String} client-prefix name.
+	 */
+	validClientPrefix(clientPrefix) {
```

Comment:
```suggestion
	validateClientPrefix(clientPrefix) {
```
Sería mas descriptivo para la función, que se llame `validateClientPrefix` en vez de `validClientPrefix`.